File: math/src/main/java/pl/edu/pg/benchmarking/math/dto/DistanceResponse.java
package pl.edu.pg.benchmarking.math.dto;

import java.util.function.Function;

public class DistanceResponse {

    private Double distance;

    public Double getDistance() {
        return distance;
    }

    private DistanceResponse(Double distance) {
        this.distance = distance;
    }

    public static Function<Double, DistanceResponse> entityToDtoMapper() {
        return DistanceResponse::new;
    }

}

File: .NET/Monolith/Models/Responses/DistanceResponse.cs
namespace Monolith.Models.Responses
{
    public class DistanceResponse
    {
        public double Distance { get; private set; }

        private DistanceResponse(double distance)
        {
            Distance = distance;
        }

        public static DistanceResponse EntityToDtoMapper(double distance)
            => new(distance);
    }
}
